app/models/user.rb

```diff
@@ -11,6 +11,19 @@ class User < ApplicationRecord
   validates :email, uniqueness: true
   validates_presence_of :first_name, :last_name
 
+  def self.create_with_omniauth(auth)
+
+    user = find_or_create_by(uid: auth['uid'], provider:  auth['provider'])
+    user.email = "#{auth['uid']}@#{auth['provider']}.com"
+    user.name = auth['info']['name']
+    if User.exists?(user)
+      user
+    else
+      user.save!
+      user
+    end
+  end
```

Comment:
I'm a person who favors keeping the model classes lean, where you can move the logic to some other patterns where you use `User` class to create a user through facebook. 

If you're interested in refactoring, you can look into something additionally: https://www.engineyard.com/blog/keeping-your-rails-controllers-dry-with-services